vector/src/main/java/im/vector/app/features/analytics/impl/DefaultVectorAnalytics.kt

```diff
@@ -41,15 +40,30 @@ private val IGNORED_OPTIONS: Options? = null
 
 @Singleton
 class DefaultVectorAnalytics @Inject constructor(
-        private val context: Context,
-        private val analyticsStore: AnalyticsStore
+        postHogFactory: PostHogFactory,
+        analyticsConfig: AnalyticsConfig,
+        private val analyticsStore: AnalyticsStore,
+        private val lateInitUserPropertiesFactory: LateInitUserPropertiesFactory,
+        @NamedGlobalScope private val globalScope: CoroutineScope
 ) : VectorAnalytics {
-    private var posthog: PostHog? = null
+
+    private val posthog: PostHog? = when {
+        analyticsConfig.isEnabled -> postHogFactory.createPosthog()
+        else                      -> {
+            Timber.tag(analyticsTag.value).w("Analytics is disabled")
+            null
+        }
+    }
 
     // Cache for the store values
     private var userConsent: Boolean? = null
     private var analyticsId: String? = null
 
+    override fun init() {
+        observeUserConsent()
```

Comment:
`createAnalyticsClient` has been removed from the custom init and inlined straight into the class field to help simplify the flow

previously `createAnalyticsClient` was also calling 

```kotlin
optOutPostHog()
identifyPostHog()
```
 
however, these calls were being made before values were set for the `userConsent` and `analyticsIds`, making them noops

```kotlin
private var userConsent: Boolean? = null
private var analyticsId: String? = null
...

// optOutPostHog
userConsent?.let { posthog?.optOut(!it) }

// identifyPostHog
val id = analyticsId ?: return


```